File: searching/js/LinearSearch.js
const search = (array, element) => {
  for (let i = 0; i < array.length; i++) {
    if (array[i] === element) return i;
  }
  return -1;
};

const array1 = [...Array(20).keys()];
const array2 = [...Array(1000000).keys()];
console.log(search(array1, 3));
console.log(search(array1, 22));
console.log(search(array2, 15832));

File: searching/ruby/linear_search.rb
def search(array, item)
    array.each_with_index { |element, idx| return idx if element == item}
    return -1
end

array_1 = (0..20).to_a
array_2 = (0..1_000_000).to_a

p search(array_1, 3)
p search(array_1, 22)
p search(array_2, 15832)

